### DISCUSSION TITLE
Error - cant open rp2030-decoder-elf

### DISCUSSION THREAD

**Discussion by marktomkins** [gab-k/RP2040-Decoder#27] [Q&A] [answered]
I've been working my way through the steps of "Getting Started" to make sure I can build the software onto a RP2040 as I have no prior experience with these RP2040's.  Anyway, after sorting various issues along the way, I have finally got to step 15.  Seems to go okay right until the last, when I get an error "Couldn't open rp2040-decoder.elf".  I've searched my Win 11 PC, and no such file exists, so clearly something has gone wrong earlier in earlier steps, although it didnt seem to.  I assume this file should have been built in step 14, so maybe thats is where the problem is; in fact I'm not sure about this step, there are two lines of code, 1st line being  cmake -G "MinGW Makefiles" -DCMAKE_BUILD_TYPE=Debug ..""  - that line seems to work okay, but then there is "mingw32".  I have tried that after the 1st line, but it throws back this :

mingw32 : The term 'mingw32' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that
the path is correct and try again.
At line:1 char:1
+ mingw32
+ ~~~~~~~
    + CategoryInfo          : ObjectNotFound: (mingw32:String) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : CommandNotFoundException

Could this be where my problem is ?

**Comment by s0len0id** [comment]
I'd guess step 10 would have taken care of being able to find the mingw32 executable (downloaded in step 9).

**Reply by gab-k** [reply]
Yeah, check your path variable and validate that the correct mingw directory is in there. Also make sure to re-open your shell window after adding it to your path variable sometimes that does the trick on windows machines. You should also verify that you have correctly installed the mingw-w64-x86_64-toolchain.

**Reply by marktomkins** [reply]
Okay, thanks for that.  So you both think the issue is with the error when I run mingw32 in step 14 I gather.  And that the most likely candidate is lack of appropriate PATH variable to point to Mingw32.  Okay, so the question I have then is, what is the correct PATH variable?  I have installed everything exactly as per guide (I haven't opted for alternative directories).  So, @s0len0id noted, in step 10 there is a PATH setting, but is does that just take care for mingw64 ?  Should I have the same syntax PATH, but for mingw32 as well?  SO it would be "C:\msys64\mingw32\bin" to add to path, or do I need to point somewhere else ?

**Reply by marktomkins** [reply]
Because I tried that (adding that path as above) and it made no difference

**Reply by s0len0id** [reply, marked as answer]
All these 32 and 64 bit msys and mingw folders are a bit confusing when troubleshooting, yes :)
But actually there appears to be typo in the "Getting Started": the executable is called **mingw32-make.exe** (with current distros, just installed to see for myself). That will produce the elf file:
```
...
[100%] Linking CXX executable RP2040-Decoder.elf
[100%] Built target RP2040-Decoder
PS C:\tmp\RP2040-Decoder\build> dir


    Directory: C:\tmp\RP2040-Decoder\build


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         22-7-2024     00:44                CMakeFiles
d-----         22-7-2024     00:43                generated
d-----         22-7-2024     00:43                pico-sdk
-a----         22-7-2024     00:43          22179 CMakeCache.txt
-a----         22-7-2024     00:43           1809 cmake_install.cmake
-a----         22-7-2024     00:43         110981 Makefile
-a----         22-7-2024     00:45         478328 RP2040-Decoder.elf
-a----         22-7-2024     00:45         315276 RP2040-Decoder.elf.map
```

**Reply by marktomkins** [reply]
Brilliant - that fixed it!  Thanks so much.  Have you built and run any of these decoders yourself ?  Any experience to share if you have ?

**Reply by marktomkins** [reply]
Oh, I did get one error in the flashing, right at the end of the operation, I got " Error: Failed to select multidrop rp2040.dapl". Anything to be concerned about ?. The rest of it went like this :

Open On-Chip Debugger 0.12.0-g4257276 (2023-01-27-10:19)
Licensed under GNU GPL v2
For bug reports, read
http://openocd.org/doc/doxygen/bugs.html
Info : Hardware thread awareness created
Info : Hardware thread awareness created
adapter speed: 5000 kHz

Info : Using CMSIS-DAPv2 interface with VID:PID=0x2e8a:0x000c, serial=E6616408433B4D21
Info : CMSIS-DAP: SWD supported
Info : CMSIS-DAP: Atomic commands supported
Info : CMSIS-DAP: Test domain timer supported
Info : CMSIS-DAP: FW Version = 2.0.0
Info : CMSIS-DAP: Interface Initialised (SWD)
Info : SWCLK/TCK = 0 SWDIO/TMS = 0 TDI = 0 TDO = 0 nTRST = 0 nRESET = 0
Info : CMSIS-DAP: Interface ready
Info : clock speed 5000 kHz
Info : SWD DPIDR 0x0bc12477, DLPIDR 0x00000001
Info : SWD DPIDR 0x0bc12477, DLPIDR 0x10000001
Info : [rp2040.core0] Cortex-M0+ r0p1 processor detected
Info : [rp2040.core0] target has 4 breakpoints, 2 watchpoints
Info : [rp2040.core1] Cortex-M0+ r0p1 processor detected
Info : [rp2040.core1] target has 4 breakpoints, 2 watchpoints
Info : starting gdb server for rp2040.core0 on 3333
Info : Listening on port 3333 for gdb connections
[rp2040.core0] halted due to debug-request, current mode: Thread
xPSR: 0xf1000000 pc: 0x000000ea msp: 0x20041f00
[rp2040.core1] halted due to debug-request, current mode: Thread
xPSR: 0xf1000000 pc: 0x000000ea msp: 0x20041f00
** Programming Started **
Info : Found flash device 'win w25q16jv' (ID 0x001540ef)
Info : RP2040 B0 Flash Probe: 2097152 bytes https://github.com/0x10000000, in 32 sectors

Info : Padding image section 1 at 0x100090a8 with 88 bytes (bank write end alignment)
Warn : Adding extra erase range, 0x10009100 .. 0x1000ffff
** Programming Finished **
** Verify Started **
** Verified OK **
** Resetting Target **
Error: Failed to select multidrop rp2040.dap1
Info : SWD DPIDR 0x0bc12477, DLPIDR 0x10000001
shutdown command invoked

**Reply by s0len0id** [reply]
no experience yet, you are 2 steps ahead, following with great interest :)

I guess if it's just the reset command that failed, all is well enough because manually power cycling will do that too?

**Reply by marktomkins** [reply]
Yeah, I did it a second time, and no error, so I guess just a power cycling thing.  Just debating if go ahead and order PCBs from JLCPCB, it's about USD 60 for 10 (fully made up and ready to go in theory).  Or do I just buy some commercial ones (US 15 each) from laisdcc on aliexpress.  hmm...
